test/trim_trailing_whitespaces_test.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require_relative 'lib'
+
+describe 'Trim trailing whitespaces tests' do
+  [
+    [' ', 'default'],
+    ['#', 'custom'],
+  ].each do |whitespace, name|
+    it "does not trim non trailing #{name} whitespaces" do
+      printer = Oppen::Wadler.new(width: 5, whitespace:)
+      printer.text("#{whitespace}a#{whitespace}")
+      printer.text("#{whitespace}b")
+      _(printer.output).must_equal "#{whitespace}a#{whitespace}#{whitespace}b"
+    end
+
+    it "trims a pure #{name} whitespace token" do
+      printer = Oppen::Wadler.new(width: 5, whitespace:)
+      printer.text(whitespace)
+      printer.break
+      _(printer.output).must_equal "\n"
+    end
+
+    it "trims multiple pure #{name} whitespace tokens" do
+      printer = Oppen::Wadler.new(width: 5, whitespace:)
+      printer.text(whitespace * 42)
+      printer.break
+      _(printer.output).must_equal "\n"
+    end
+
+    it "trims a single #{name} trailing whitespace" do
+      printer = Oppen::Wadler.new(width: 5, whitespace:)
+      printer.text("#{whitespace}a#{whitespace}")
+      printer.break
+      printer.text("#{whitespace}b")
+      _(printer.output).must_equal <<~LANG.chomp
+        #{whitespace}a
+        #{whitespace}b
+      LANG
+    end
+
+    it "trims multiple #{name} trailing whitespaces" do
+      printer = Oppen::Wadler.new(width: 5, whitespace:)
+      printer.text("#{whitespace}a#{whitespace * 42}")
+      printer.break
+      printer.text("#{whitespace}b")
+      _(printer.output).must_equal <<~LANG.chomp
+        #{whitespace}a
+        #{whitespace}b
+      LANG
+    end
+
+    it "trims #{name} whitespace of previous parent group" do
+      printer = Oppen::Wadler.new(width: 5, whitespace:)
+      printer.text("a#{whitespace}")
+      printer.group {
+        printer.group {
+          printer.break
+          printer.text('b')
+        }
+      }
+      _(printer.output).must_equal <<~LANG.chomp
+        a
+        b
+      LANG
+    end
+
+    it "trims #{name} whitespace of previous child group" do
+      printer = Oppen::Wadler.new(width: 5, whitespace:)
+      printer.group {
+        printer.group {
+          printer.group {
+            printer.text("a#{whitespace}")
+          }
+        }
+        printer.break
+        printer.text('b')
+      }
+      _(printer.output).must_equal <<~LANG.chomp
+        a
+        b
+      LANG
+    end
```

Comment:
We also need the following cases:

1. text, whitespace(s), break(s)
1. text, whitespace(s), break(s), whitespaces
1. text, whitespace(s), break(s), whitespaces, break(s)
1. text, whitespace(s), break(s), whitespaces, break(s), whitespaces